packages/core/src/shared/telemetry/util.ts

```diff
@@ -90,6 +91,37 @@ export function convertLegacy(value: unknown): boolean {
     }
 }
 
+/**
+ * Returns an identifier that uniquely identifies a single application
+ * instance/window of a specific IDE. I.e if I have multiple VS Code
+ * windows open each one will have a unique session ID. This session ID
+ * can be used in conjunction with the client ID to differntiate between
+ * different VS Code windows on a users machine.
+ *
+ * ### Rules:
+ *
+ * - On startup of a new application instance, a new session ID must be created.
+ *   - This identifier must be a `UUID`, it is enforced by the telemetry service.
+ * - The session ID must be different from all other IDs on the machine
+ * - A session ID exists until the application instance is terminated.
+ *   It should never be used again after termination.
+ * - All extensions on the same application instance should return the same session ID.
+ *
+ * `vscode.env.sessionId` behaves as described aboved, EXCEPT its
+ * value looks close to a UUID by does not exactly match it (has additional characters).
```

Comment:
Why does it matter what it looks like if its a unique and random value?

Reply:
The server enforces it, mentioned here: https://github.com/aws/aws-toolkit-vscode/pull/5510/files#diff-6b731a821720f9104e0c179a1bc64d1039af6d386549d43090cc5c661be0f79eR104

Reply:
> We cannot guarantee the format of vscode.env.sessionId as it has no spec, at any point it could change to something else. Then the server would need to modify the pattern it accepts. UUID is a widely recognized format and is something we can consistently guarantee.

> The other IDEs will use this feature as well, if they have something similar to vscode.env.sessionId but its format is slightly different from ours, then should we modify the server to also accept that? I'd rather us have an obvious interface of requiring a UUID that all IDEs can easily follow.

Make the format any string and let the client determine what a "session ID" is, we should be accommodating to existing tools. The format validation isn't useful here - it doesn't stop clients from sending non-unique ids (and might make it harder to detect this if it "looks" right). 
Also, if there is a bug where clients send non-unique IDs, what is the outcome? It is not catastrophic like non-unique `clientId`